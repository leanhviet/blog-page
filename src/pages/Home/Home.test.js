// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { HomeBase } from './index'

const props = {
  mainData: {
    blogs: [
      {
        id: 1,
        content: 'This is content',
        title: 'Title',
      },
      {
        id: 2,
        content: 'This is content 2',
        title: 'Title 2',
      },
    ],
    totalRecords: 30,
  },
}

describe('HomeBase component', () => {
  const wrapper = shallow(<HomeBase />)
  const wrapperWithProps = shallow(<HomeBase {...props} />)

  it('should be defined HomeBase component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot HomeBase component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps HomeBase component', () => {
    expect(HomeBase.defaultProps.mainData).toEqual({
      blogs: [],
      totalRecords: 0,
    })
  })

  // Testing mock props
  it('should be render mockProps HomeBase component', () => {
    expect(wrapperWithProps.find('Memo(BlogItemBase)').length).toBe(2)
  })
})
