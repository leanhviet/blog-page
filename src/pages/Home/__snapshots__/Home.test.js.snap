// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeBase component should be render snapshot HomeBase component 1`] = `
<Container
  className="border home"
  fluid={false}
>
  <HeaderBase />
  <Memo(FeatureGroupBase)
    handleSearch={[Function]}
    handleSortBy={[Function]}
  />
</Container>
`;

exports[`HomeBase component should be render snapshot HomeBase component 2`] = `
<Container
  className="border home"
  fluid={false}
>
  <HeaderBase />
  <Memo(FeatureGroupBase)
    handleSearch={[Function]}
    handleSortBy={[Function]}
  />
  <Memo(BlogItemBase)
    blog={
      Object {
        "content": "This is content",
        "id": 1,
        "title": "Title",
      }
    }
    key="1"
  />
  <Memo(BlogItemBase)
    blog={
      Object {
        "content": "This is content 2",
        "id": 2,
        "title": "Title 2",
      }
    }
    key="2"
  />
  <Memo(PaginationBase)
    initialPage={1}
    onChangePage={[Function]}
    pageLimit={3}
    pagesToShow={3}
    totalRecords={30}
  />
</Container>
`;
